### ISSUE TITLE
Docker containers should not include a poetry cache

### ISSUE THREAD

**Issue by gsnider2195** [nautobot/nautobot#2342]
### Proposed Changes

Run `poetry cache clear -n --all pypi` in the `final` step of the Dockerfile

### Justification

Reduce file size of docker container by removing cached wheel files that are no longer needed

**Comment by glennmatthews** [comment]
Appears to be currently blocked by https://github.com/python-poetry/poetry/pull/6212
